wallet-api/src/Wallet/API.hs

```diff
@@ -71,11 +165,30 @@ data WalletAPIError =
 class WalletAPI m where
     submitTxn :: Tx -> m ()
     myKeyPair :: m KeyPair
+
+    {- |
+    Create a payment that spends the specified value and returns any
+    leftover funds as change. Fails if we don't have enough funds.
+    -}
     createPaymentWithChange :: Value -> m (Set.Set TxIn', TxOut')
-    -- ^ Create a payment that spends the specified value and returns any
-    --   leftover funds as change. Fails if we don't have enough funds.
+
+    {- |
+    Register an action in `m ()` to be run when condition is true.
+
+    * The action will be run once for each block where the condition holds. For
+      example, `register (blockHeightT (Interval 3 6)) a` causes `a` to be run
+      at blocks 3, 4, and 5.
+
+    * Each time the wallet is notified of a new block, all triggers are checked
+      and the matching ones are run in an unspecified order.
+
+    * The wallet will only watch addresses that have a trigger registered for
+      them. Therefore it only counts funds that are added or removed *after*
+      the trigger has been registered.
```

Comment:
> so the funds-at trigger will be based on a state that doesn't really reflect the chain at all?

Yes, it's not ideal, but the other option is to watch all script addresses all the time. And it does reflect the chain in some sense, because we know that the amount in our watch list is smaller than or equal to the true amount (so it's not an arbitrary relation)

Reply:
Well, the wallet could query the chain for the real information when you set up the trigger. Perhaps that's wallet functionality we might actually want?

Reply:
That would still require a scan of the entire blockchain. I don't think the performance of `register` should depend on the size of the chain at the time it is called. (It only affects the user who is running the contract, but it still seems undesirable)

Reply:
Well, I don't see how you can avoid it if you want to write a predicate that depends on such global state!

The current version just doesn't seem that useful to me, because it doesn't reflect the true balance.

Reply:
Following the call earlier today, I'll change the wallet API implementation to automatically watch all addresses in transaction outputs that a wallet submits to the chain.